src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp

```diff
@@ -252,8 +252,11 @@ namespace aiebu {
           auto bw_header = reinterpret_cast<const XAie_BlockWrite32Hdr *>(ptr);
           auto payload = reinterpret_cast<const char*>(ptr + sizeof(XAie_BlockWrite32Hdr));
           auto offset = static_cast<uint32_t>(payload-mc_code.data());
-          uint64_t buffer_length_in_bytes = reinterpret_cast<const uint64_t*>(payload)[0] * 4;
-          blockWriteRegOffsetMap[bw_header->RegOff] = std::make_pair(offset, buffer_length_in_bytes);
+          uint32_t size = (bw_header->Size - sizeof(*bw_header));
+          for (auto bd = 0U ; bd < size; bd+=SHIM_DMA_BD_SIZE) { //size and bd in bytes
```

Comment:
Can you add a few lines of comment to describe the operation here?

Reply:
added